crates/xline-client/examples/kv.rs

```diff
@@ -17,8 +17,8 @@ async fn main() -> Result<()> {
         .kv_client();
 
     // put
-    client.put(PutRequest::new("key1", "value1")).await?;
-    client.put(PutRequest::new("key2", "value2")).await?;
+    client.put("key1", "value1", None).await?;
+    client.put("key2", "value2", None).await?;
```

Comment:
Can we implement a builder pattern that supports:
```rust
client.put("key1", "value1").await?;
client.put("key1", "value1").with_prev_key().await?;
```
Then we don't need to explicitly write a `PutOptions`.

Reply:
That sounds good! But how can we implement like that?
Suppose the `put` function returns a struct `PutRequest`, do we need 
1. `impl Future for PutRequest` to make it awaitable?
2. `impl PutRequestBuilder` with an `async fn execute()`, and call it as `client.put("key1", "value1").with_prev_key().execute().await?;`
3. [using traits](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=62c2dff987602f3a6f1b6b6cefaf4616), but needs to record a `prev_key` state in Client member.

Reply:
I think implement a future could be the most convenient. Maybe something like: 
```rust
struct PutFut {
    inner: PutRequest,
}

impl Future for PutFut {
    // send the command
}

impl PutFut {
    fn with_prev_key(mut self) -> Self {
        self.inner.prev_key = true;
        self
    }
}

impl Client {
    fn put(&self, key: Vec<u8>, value: Vec<u8>) -> PutFut {
        ...
    }
}


```

Reply:
There're some questions:
1. Since we have to send command in Future impl, we need to pass `curp_client` and `token` to `PutFut`
2. We need to construct command in `poll()`. `poll()` will be called multiple times, the command needs to be stored somewhere. But `Self` has a mut ref (in `poll()` parameter), so we cannot store it in `PutFut`. ([ex code here](https://gist.github.com/lxl66566/2d39e1d3efc8715cb77ce33dafa8d301), `cmd` does not live long enough).
3. I don't know how to modify the `TxnOp::put` impl below

Reply:
For 2 you can move the command into the future
```rust
            let cmd = Command::new(RequestWrapper::from(self.inner.take().unwrap()));
            let client = Arc::clone(self.curp_client);
            let token = self.token.cloned();
            let fut = async move { client.propose(&cmd, token.as_ref(), true).await };
            self.fut = Some(Box::pin(fut));
 ```

Reply:
For 3, maybe:
```rust
    client.txn().when(...).and_then(|c| {
        c.put("key0", "value0").with_prev_key();
        c.put("key1", "value1");
    }).await?;
```

Reply:
If you find it too complex to implement, you can always opt for the original approach.